### PR TITLE
[CCI] Update js-yaml to v4.0.5

### PR DESCRIPTION
**Pull request by andreymyssak**
### Description
Update `js-yaml` to v4.0.5. The update went smoothly, without any major changes.

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3659
 
### Check List
- [X] All tests pass
  - [X] `yarn test:jest`
  - [X] `yarn test:jest_integration`
  - [X] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [X] Update [CHANGELOG.md](./../CHANGELOG.md)
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by andreymyssak** [comment]
@kavilla In which section of `CHANGELOG.md` should I put a line about this change?

**Comment by ashwin-pc** [comment]
@andreymyssak Id put it under the `Maintenance` category

File: packages/osd-apm-config-loader/src/utils/read_config.ts [hunk 1]
```diff
@@ -29,13 +29,13 @@
  */
 
 import { readFileSync } from 'fs';
-import { safeLoad } from 'js-yaml';
+import { load } from 'js-yaml';
```
Comment (joshuarrrr) [inline packages/osd-apm-config-loader/src/utils/read_config.ts:32]: @andreymyssak Can you explain this change a little? Is it just a method renaming, or are there functional differences between the behavior of the old `safeLoad` vs `load`?

Comment (andreymyssak) [reply packages/osd-apm-config-loader/src/utils/read_config.ts:32]: Just renaming, see [safeLoad(), safeLoadAll(), safeDump() => load(), loadAll(), dump()](https://github.com/nodeca/js-yaml/blob/master/migrate_v3_to_v4.md#safeload-safeloadall-safedump--load-loadall-dump). There's no mention of a breaking change in their behavior.

Comment (joshuarrrr) [reply packages/osd-apm-config-loader/src/utils/read_config.ts:32]: perfect, thanks for the link!

**Review by joshuarrrr — APPROVED** [review]

**Review by manasvinibs — APPROVED** [review]
Thanks for the fix. LGTM

**Comment by AMoo-Miki** [comment]
This will impact plugins so no backporting?

**Comment by joshuarrrr** [comment]
> This will impact plugins so no backporting?

Correct, this is not backportable as-is. Labeled for 3.0.0.

**Review by manasvinibs — APPROVED** [review]

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**